### DISCUSSION TITLE
About the compute offers

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#20701] [Questions] [answered]
I could not find clarification of the compute add-on meaning, mainly the two lines "direction connections" and "pooler connections"
![image](https://github.com/supabase/supabase/assets/17619412/8a715d11-4c4d-40c3-a1ca-41c5831fbce2)

does that mean in this example that Micro compute can only accepts 60 realtime connections for example?
or is it completely different?

**Comment by GaryAustin1** [comment]
It is completely different. Realtime all goes thru a set of global realtime servers.  Realtime quotas are here: https://supabase.com/docs/guides/realtime/quotas .  This https://supabase.com/docs/guides/realtime/postgres-changes#limitations also gives you some ideas on realtime performance on different size instances.

The connections there are the recommended max number of database connections from your apps.

Edit: Note if you come in thru the REST API those are not directly relevant to number of users as the connections are shared by a pooler in the API itself.

**Reply by GaryAustin1** [reply, marked as answer]
Continuing in same "answer"...
I edited to say if you are using the REST API those are not directly relevant to you.  The database API (postgres-js) for instance all goes thru its own pooler.  Number of users is not capped.  It is up to how complex their calls are and how often they make them that will limit you mainly from CPU/memory etc.  But it is not a fixed number of users is the main thing.

The connections are mainly relevant for things like Prisma that use direct database connections.

**Reply by pierroo** [reply]
Thank you for this, now I understand better this specific part.
Now I am only worried about the "10 DB changes/seconds limit" for 1000ccu+ with RLS, moment of truth shall be saturday when I temporarily open the gate of this beta feature. :')

**Reply by GaryAustin1** [reply]
Yes, that could be a concern for Realtime.  It is not a fixed limit, but what they estimate the various size instances can handle reasonably.

**Comment by pierroo** [comment]
But that's the part I don't understand "recommended max number of database connections from your apps"
=> what does it actually mean?
How do we define one database connection?

I have a mobile app.
I have 1000 concurrent users for example.
Each one of them creates a supabase client to run all their supabase queries through supabase js sdk

where is the "database connection" here?
